src/main/java/net/minecraftforge/client/ItemOverlayHandler.java

```diff
@@ -0,0 +1,123 @@
+package net.minecraftforge.client;
+
+import java.util.List;
+
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.Tessellator;
+import net.minecraft.client.renderer.VertexBuffer;
+import net.minecraft.client.renderer.block.model.BakedQuad;
+import net.minecraft.client.renderer.block.model.IBakedModel;
+import net.minecraft.client.renderer.texture.TextureManager;
+import net.minecraft.client.renderer.texture.TextureMap;
+import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumFacing;
+import net.minecraftforge.common.ForgeModContainer;
+
+/**
+ * Offshore handler for the Items' extra passes. Specifically for reducing patch size & maintaining readability.
+ * */
+public final class ItemOverlayHandler
+{
+
+    private static final java.util.Map<net.minecraftforge.fml.common.registry.RegistryDelegate<Item>, IItemOverlay> itemOverlayMap = com.google.common.collect.Maps.newHashMap();
```

Comment:
That would be useful. Also can you make the default value of the getter return `IItemOverlay.VANILLA` and never null? That would make it simpler.